core/src/main/java/com/linecorp/armeria/common/RequestContextAwareScheduledExecutorService.java

```diff
@@ -16,43 +16,95 @@
 
 package com.linecorp.armeria.common;
 
+import static java.util.Objects.requireNonNull;
+
 import java.util.concurrent.Callable;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.TimeUnit;
 
-final class RequestContextAwareScheduledExecutorService extends RequestContextAwareExecutorService
-        implements ScheduledExecutorService {
+import com.google.common.base.MoreObjects;
+
+/**
+ * A delegating {@link ScheduledExecutorService} that sets the {@link RequestContext} before executing any
+ * submitted tasks.
+ */
+public final class RequestContextAwareScheduledExecutorService
```

Comment:
I think this class name is way too long.
Can we rename classes that start with `RequestContextAware` to `ContextAware`?

Reply:
SGTM